layers/+misc/smart-tabs/packages.el

```diff
@@ -0,0 +1,41 @@
+;;; packages.el --- smart-tabs layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
+;;
+;; Author: Curtis Mackie <curtis@mackie.ninja>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary:
+
+;; This layer integrates smart-tabs-mode with Spacemacs, allowing users to indent
+;; with tabs and align with spaces. It ensures that it works as expected with
+;; evil's = operator, as well as streamlining the configuration process for new
+;; language support.
+
+;;; Code:
+
+(defconst smart-tabs-packages
+  '(smart-tabs-mode))
+
+(defun smart-tabs/init-smart-tabs-mode ()
+  (use-package smart-tabs-mode
+    :diminish smart-tabs-mode
+    :config
+    (progn
+      (add-hook 'smart-tabs-mode-hook
+                (lambda ()
```

Comment:
it's good practice to name functions that go into a hook. Instead of:
```elisp
(add-hook <some-hook> (lambda () ...))
```
Is better to use:
```elisp
(defun whatever () ...)
(add-hook <some-hook> #'whatever)
```